test/unit/test_serialize_config.c

```diff
@@ -50,9 +50,35 @@ bool test_config_load() {
 	mu_end;
 }
 
+bool test_config_load_exclude() {
+	static const char * const exclude[] = {
+		"somestring",
+		"someint",
+		NULL
+	};
+	RzConfig *config = rz_config_new (NULL);
```

Comment:
what about making the `exclude` list a constructor argument so that a `RzConfig` can be created with different exclude-lists based on the need? Maybe also a setter, so Cutter can take `core->config` and apply his own list of things he does not want to save. Do I understand it right that, if not saved, the default/current value is applied?

Reply:
No sorry, that's actually not the best thing probably... Such configuration belongs to the serialization process, not to the single classes. Maybe we should put it into `RzSerializeResultInfo`?

Reply:
Not sure what exactly you mean by putting it into `RzSerializeResultInfo` but this struct is purely for returning the result.

Reply:
You definitely know more about the serialization stuff, but IMHO the list of things you do not want to serialize is a property of the "serialization process". It's not something of `RzConfig` but not even something that `RzCore` should choose on its own.

Right now the output of the serialization process is just passed as a single `Sdb`. Maybe this should actually be `RzSerialization` which includes an `Sdb *` and some configuration fields (e.g. in this case the list of config variables to exclude from the serialization/deserialization). WDYT?

Reply:
> the list of things you do not want to serialize is a property of the "serialization process". It's not something of `RzConfig` but not even something that `RzCore` should choose on its own.

Yes. This is why the exclusion list here is currently an argument of `rz_serialize_config_load()` instead of for example a member of `RzConfigNode`.
I also agree that core should rather not choose this at `RzCore` creation time, but I think the right place is where the `RzConfig` instance is loaded in the context of a project, which is here: https://github.com/rizinorg/rizin/pull/223/files#diff-e10dc9ab50cb92db60c5eaf3dd32a456141636e478a2e6b320f25d3c28892e08R57


> Right now the output of the serialization process is just passed as a single `Sdb`. Maybe this should actually be `RzSerialization` which includes an `Sdb *` and some configuration fields (e.g. in this case the list of config variables to exclude from the serialization/deserialization). WDYT?

Yeah I think it can make sense to have such a struct that holds possibly also multiple configuration fields for the (de)serialization. But I would rather keep it clearly separated from what is the output and what is input/config. So until now we only have one output and one config, which I think is good enough as just two args.

Reply:
Okok, makes sense. Let's consider this if configuration stuff becomes more though.